app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
-  has_many :comments
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
   has_many :likes
+  has_many :comments
+
+  after_save :updates_user_posts_counter
+
+  def updates_user_posts_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
[OPTIONAL]
- [ ] Same for the method `updates_user_posts_counter` it would be nice to make it private, since it will be call every time that a new user has been save in the database.